filebeat/tests/system/test_modules.py

```diff
@@ -95,14 +97,21 @@ def run_on_file(self, module, fileset, test_file, cfgfile):
         objects = [o["_source"] for o in res["hits"]["hits"]]
         assert len(objects) > 0
         for obj in objects:
-            self.assert_fields_are_documented(obj)
-            # assert "error" not in obj  # no parsing errors
-            assert obj["fileset"]["module"] == module
+            assert obj["fileset"]["module"] == module, "expected fileset.module={} but got {}".format(
+                module, obj["fileset"]["module"])
+
+            if not (module == "mysql" and fileset == "slowlog") and not (module == "system" and fileset == "auth"):
+                # TODO: There are errors parsing the test logs from these modules.
+                assert "error" not in obj
+
+            if module != "system" and fileset != "audit":
+                # There are dynamic fields in audit logs that are not documented.
```

Comment:
If there was an `exclude_keys` option in the KV processor then it would easy to prevent the dynamic fields from overwriting the well-known fields from the header. The Logstash KV filter has this option so I'll [ask](https://github.com/elastic/elasticsearch/issues/23856) about getting this in ES. I could enhance this module to use that once it's available.

There are only a few keys in the audit header that could be overwritten (`record_type` and `sequence`) by KV data and I haven't observed these being used in any audit events. I think the risk is minimal.